cpp/src/arrow/util/utf8.cc

```diff
@@ -79,5 +79,31 @@ void InitializeUTF8() {
   std::call_once(utf8_initialized, internal::InitializeLargeTable);
 }
 
+static const uint8_t kBOM[] = "\xef\xbb\xbf";
```

Comment:
string to `uint8_t[]` seems odd (and the -1 in the loop), why not use `static const uint8_t kBOM[] = [0xEF, 0xBB, 0xBF]`?

Reply:
Ah, right, I hadn't thought about that.